tls/s2n_connection.h

```diff
@@ -153,9 +153,19 @@ struct s2n_connection {
 
     /* Maximum outgoing fragment size for this connection. Does not limit
      * incoming record size.
+     *
+     * This value is updated when:
+     *   1. s2n_connection_prefer_low_latency is set
+     *   2. s2n_connection_prefer_throughput is set
+     *   3. TLS Maximum Fragment Length extension sets mfl_code
```

Comment:
nitpick: I would change "sets mfl_code" to "is negotiated" and remove the reference to an internal flag